js/classes/entities/foreground/weather_system.js

```diff
@@ -0,0 +1,182 @@
+     */
+    makeHeavens() {
+        if (this.time === "night") {
+            BG_COLOR = COLORS.SKY_DARK_GREY;
+            GAME.addEntity(new Sun(new Vector(Camera.SIZE.x - 2 * Sun.SCALED_SIZE, Sun.SCALED_SIZE - 100), Sun.MOON), -1);
+
+        } else {
+            switch (this.type) {
+                case "warm":
+                    BG_COLOR = COLORS.SKY_BLUE;
+                    GAME.addEntity(new Sun(new Vector(Camera.SIZE.x - 2 * Sun.SCALED_SIZE, Sun.SCALED_SIZE - 100), Sun.VILLAGE), -1);
+                    break;
+                case "hot":
+                    BG_COLOR = COLORS.SKY_HOT_SKY;
+                    GAME.addEntity(new Sun(new Vector(Camera.SIZE.x - 2 * Sun.SCALED_SIZE, Sun.SCALED_SIZE - 100), Sun.LAVA), -1);
+                    break;
+                case "cloudy":
+                    BG_COLOR = COLORS.SKY_GREY;
+                    break;
+                case "rain":
+                    BG_COLOR = COLORS.SKY_GREY;
+                    break;
+                case "snow":
+                    BG_COLOR = COLORS.SKY_SNOW_GREY;
+                    break;
+                default:
+                    break;
+            }
+        }
+
+    }
+
+    /**
+     * Adds clouds.
+     */
+    makeClouds() {
+        // Make sure to only have a number of clouds that is divisble by 100. For example, 15 is bad value but 25 is a good value.
+        const numOfClouds = [2, 5, 10, 20, 25, 50];
+        let aboveGroundLevel = 19;
+        // TODO add in dark clouds.
+        if (this.makeRain) {
+            // Add dark clouds
+        } else {
+            // Add white clouds
+        }
+
+        let cloudNum = numOfClouds[this.intensity];
+        // Make so many clouds of each type.
+        for (let cloudType = 0; cloudType < 3; cloudType++) {
+            // Choose a multiplier that spaces out the clouds across the whole zone nicely.
+            let spaceBetweenMultiplier = 0;
+            for (spaceBetweenMultiplier = 0; spaceBetweenMultiplier < 100; spaceBetweenMultiplier++) {
+                if (spaceBetweenMultiplier * cloudNum == 100) {
+                    break;
+                }
+            }
+
+            for (let i = 0; i < cloudNum; i++) {
+                let foreground = 0;
+                // Give the clouds some random variation.
+                let yVariation = Math.random() * (10 - 16) + 14;
+                let xVariation = Math.random() * (20 - 50) + 20;
+                // Choose a location for each cloud that is added such that it has a random x variation and it is spaced out over the zone.
+                // i * spaceBetweenMultiplier ensure that there will be clouds across the whole zone.
+                let randomOrigin = xVariation + i * spaceBetweenMultiplier;
+                let cloudPosition = Vector.blockToWorldSpace(new Vector(randomOrigin, aboveGroundLevel - yVariation));
+                let cloudVarient = Decoration.DECORATIONS.clouds.CLOUD_BUSHY;
```

Comment:
Typo: `cloudVarient` -> `cloudvariant`... That said, generally, you use the word type when describing a variant, so perhaps `cloudType`.

Reply:
Good idea, thanks for pointing that out!